OpenRA.Game/MiniYaml.cs

```diff
@@ -215,64 +243,84 @@ static List<MiniYamlNode> FromLines(IEnumerable<string> lines, string filename,
 					// Leading and trailing whitespace is always trimmed from keys.
 					// Leading and trailing whitespace is trimmed from values unless they
 					// are marked with leading or trailing backslashes
-					var keyLength = line.Length - keyStart;
+					var keyLength = lineLength - keyStart;
 					var valueStart = -1;
 					var valueLength = 0;
 					var commentStart = -1;
-					for (var i = 0; i < line.Length; i++)
+					var hasEscapedHashChar = false;
+					var prevChar = '\0';
+					for (var i = keyStart; i < lineLength; i++)
 					{
-						if (valueStart < 0 && line[i] == ':')
+						var currChar = line[i];
+						if (commentStart < 0)
 						{
-							valueStart = i + 1;
-							keyLength = i - keyStart;
-							valueLength = line.Length - i - 1;
-						}
+							if (currChar == '#')
+							{
+								if (i == 0 || prevChar != '\\')
+								{
+									commentStart = i + 1;
+									if (commentStart <= keyLength)
+										keyLength = i - keyStart;
+									else
+										valueLength = i - valueStart;
 
-						if (commentStart < 0 && line[i] == '#' && (i == 0 || line[i - 1] != '\\'))
-						{
-							commentStart = i + 1;
-							if (commentStart <= keyLength)
+									break;
+								}
+								else
+									hasEscapedHashChar = true;
+							}
+							else if (valueStart < 0 && currChar == ':')
+							{
+								valueStart = i + 1;
 								keyLength = i - keyStart;
-							else
-								valueLength = i - valueStart;
+								valueLength = lineLength - i - 1;
```

Comment:
Is it intentional that we only set this in case there is a comment on the line now?

Reply:
How do you mean?

This is pretty much the original source code. We set it when there is a comment above and here when there is no comment.

We set it to avoid having to find the end after once more using an "expensive" string operation.

Reply:
The code looks fine to me now, I must have overlooked something yesterday.